enterprise/server/auth/auth.go

```diff
@@ -38,6 +66,129 @@ func RegisterNullAuth(env environment.Env) error {
 	return nil
 }
 
+type authenticator struct {
+	http         []interfaces.HTTPAuthenticator
+	user         []interfaces.UserAuthenticator
+	grpc         interfaces.GRPCAuthenticator
+	apiKey       interfaces.APIKeyAuthenticator
+	installation interfaces.InstallationAuthenticator
```

Comment:
nit: if you want, you could embed the interfaces instead of having them as fields, to avoid having to define the "passthrough" methods that delegate directly

i.e.

```
type authenticator struct {
  interfaces.GRPCAuthenticator
}

auth := &authenticator{
  ...,
  GRPCAuthenticator: oidc,
}

// can delete AuthenticatedGRPCContext() now
```

Reply:
Fancy, done